src/Components/SavedSearchAlert/Components/SavedSearchAlertNameInput.tsx

```diff
@@ -0,0 +1,101 @@
+import { Input } from "@artsy/palette"
+import { SavedSearchAlertFormValues } from "Components/SavedSearchAlert/types"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { useFormikContext } from "formik"
+import { FC, useEffect, useMemo, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { SavedSearchAlertNameInputQuery } from "__generated__/SavedSearchAlertNameInputQuery.graphql"
+import { useSavedSearchAlertContext } from "Components/SavedSearchAlert/SavedSearchAlertContext"
+import { debounce } from "lodash"
+import { SavedSearchAlertNameInput_previewSavedSearch$data } from "__generated__/SavedSearchAlertNameInput_previewSavedSearch.graphql"
+import { useFeatureFlag } from "System/useFeatureFlag"
+
+export const SavedSearchAlertNameInputQueryRenderer: FC = () => {
+  const { criteria } = useSavedSearchAlertContext()
+  const [criteriaState, setCriteriaState] = useState({})
+
+  const debouncedSetCriteriaState = useMemo(
+    () => debounce(setCriteriaState, 200),
+    []
+  )
+
+  useEffect(() => {
+    debouncedSetCriteriaState(criteria)
+  }, [criteria, debouncedSetCriteriaState])
+
+  return (
+    <SystemQueryRenderer<SavedSearchAlertNameInputQuery>
+      query={graphql`
+        query SavedSearchAlertNameInputQuery(
+          $attributes: PreviewSavedSearchAttributes!
+        ) {
+          previewSavedSearch(attributes: $attributes) {
+            ...SavedSearchAlertNameInput_previewSavedSearch
+          }
+        }
+      `}
+      variables={{ attributes: criteriaState }}
+      render={({ props, error }) => {
+        if (error) {
+          console.error(error)
+        }
+
+        return (
+          <SavedSearchAlertNameInputFragmentContainer
+            // @ts-expect-error "No overload matches this call" TODO: why?
+            previewSavedSearch={props?.previewSavedSearch}
```

Comment:
It seems like the fragment container expects `previewSavedSearch` to not be undefined.
```
Warning: createFragmentSpecResolver: Expected prop `previewSavedSearch` to be supplied to `Relay(SavedSearchAlertNameInput)`, but got `undefined`. Pass an explicit `null` if this is intentional. 
```

Now even passing `null` results in a type error 🤔

Reply:
To solve the issue, we could change the line to 

```suggestion
            // @ts-expect-error "No overload matches this call" TODO: why?
            previewSavedSearch={(props?.previewSavedSearch ?? null)!}
```

 or remove the fragment container and add the field `displayName` to the query and pass the value directly to `SavedSearchAlertNameInput` like this:

```suggestion
            // @ts-expect-error "No overload matches this call" TODO: why?
            displayName={props?.previewSavedSearch?.displayName}
```